Notes toward support for pure Swift

loganj opened the pull request:
This is an exploration of adding pure swift support to sqldelight. Might be a thing?

loganj commented on pure-swift-notes/pure-swift-notes-build.md:
I had notes on bazel integration, but lost them while learning how to iterate well with goose/claude (small files, not big ones, for sure).

Need to re-tread that ground.

zadr commented on pure-swift-notes/pure-swift-notes-concurrency.md:
```diff
@@ -0,0 +1,324 @@
+    }
+}
+
+// Option B: Actor-based isolation
```
its gotta be this one. dispatch queues are the older non-coroutine-style API

loganj replied:
Heard! Thank you

loganj committed:
move most of the compiler's kotlin backend into compiler.kotlin

loganj committed:
add swiftpoet dependency

loganj committed:
Tease TargetType/KotlinType out of DialectType

DialectType now just represents the type in the SQL dialect, and can produce a
target type. Current implementations secretly implement both DialectType and
KotlinType – when asked as DialectTypes to produce a KotlinType, they produce
themselves.  We'll actually break that down when we start having DialectTypes
also produce SwiftTypes.

loganj committed:
Extract extensions from NamedQuery and BindableQuery into compiler/kotlin/Queries.kt

This removes all direct dependencies on kotlinpoet from the
core.compiler.model, and moves them to the kotlin backend package.

loganj committed:
introduce SwiftType

loganj committed:
PrimitiveType no longer implements KotlinType directly

Making room for SwiftType

loganj committed:
introduce SwiftPrimitive

loganj committed:
Spike SwiftTableInterfaceGenerator

loganj committed:
Add SqlDelightCLI

We would like a CLI for invoking the compiler, along the lines of square/sqldelight_bazel_rules

This necessitated moving SqlDelightEnvironment into sqlidelight-compiler, as
the CLI needs to use it.  The gradle plugin already depends on this module, so
seems fine?

loganj committed:
There might not be a parent directory

loganj committed:
Extract KotlinBackend and stub a SwiftBackend

loganj committed:
add CLI support for selecting backend

loganj committed:
Less than ideal implementation of SwiftBackend.needsWrapper

loganj committed:
Spike in a Swift runtime
